mysql-test/suite/innodb/t/percona_force_encryption.test

```diff
@@ -0,0 +1,92 @@
+create table t4_on_explicit_off (a int) engine=innodb encryption="N";
+set global innodb_encrypt_tables='FORCE';
+--error ER_INVALID_ENCRYPTION_OPTION
+create table t5_force_explicit_off (a int) engine=innodb encryption="N";
+create table t6_force_default (a int) engine=innodb;
+set global innodb_encrypt_tables='OFF';
+create table t5_off_explicit_off (a int) engine=innodb encryption="N";
+set global innodb_encrypt_tables='FORCE';
+
+insert into t1_default_explicit_on values (42);
+insert into t2_default_explicit_off values (42);
+insert into t3_on_default values (42);
+insert into t4_on_explicit_off values (42);
+insert into t5_off_explicit_off values (42);
+insert into t6_force_default values (42);
+
+show create table t1_default_explicit_on;
+show create table t2_default_explicit_off;
+show create table t3_on_default;
+show create table t4_on_explicit_off;
+show create table t5_off_explicit_off;
+show create table t6_force_default;
+
+--error ER_INVALID_ENCRYPTION_OPTION
+alter table t1_default_explicit_on encryption="N";
+alter table t2_default_explicit_off encryption="Y";
+--error ER_INVALID_ENCRYPTION_OPTION
+alter table t2_default_explicit_off encryption="N";
+alter table t3_on_default;
+--error ER_INVALID_ENCRYPTION_OPTION
+alter table t4_on_explicit_off encryption="N";
+
+set global innodb_encrypt_tables='ON';
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table t1_default_explicit_on encryption="N", ALGORITHM=INPLACE;
+alter table t1_default_explicit_on encryption="Y", ALGORITHM=INPLACE;
+alter table t1_default_explicit_on encryption="N";
+alter table t2_default_explicit_off encryption="N";
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table t3_on_default encryption="N", ALGORITHM=INPLACE;
+alter table t3_on_default encryption="Y", ALGORITHM=INPLACE;
+alter table t3_on_default encryption="N";
+alter table t4_on_explicit_off ALGORITHM=INPLACE;
+alter table t4_on_explicit_off encryption="N";
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table t6_force_default encryption="N", ALGORITHM=INPLACE;
+
+# Even when it's ON or FORCE, non encrypted tables can be altered,
+# without changing their encryption settings
+alter table t4_on_explicit_off change a b int;
+set global innodb_encrypt_tables='FORCE';
+alter table t4_on_explicit_off change b a int;
+
+show create table t1_default_explicit_on;
+show create table t2_default_explicit_off;
+show create table t3_on_default;
+show create table t4_on_explicit_off;
+show create table t5_off_explicit_off;
+show create table t6_force_default;
+
+select * from t1_default_explicit_on;
+select * from t2_default_explicit_off;
+select * from t3_on_default;
+select * from t4_on_explicit_off;
+select * from t5_off_explicit_off;
+select * from t6_force_default;
+
+drop table t1_default_explicit_on;
+drop table t2_default_explicit_off;
+drop table t3_on_default;
+drop table t4_on_explicit_off;
+drop table t5_off_explicit_off;
+drop table t6_force_default;
+
```

Comment:
Please, also add tests for how `CREATE TABLE LIKE <existing_table>` works for different combinations of `innodb_encrypt_tables` and encryption attribute of `<existing_table>`.

Reply:
I added two new tables, their name ending with _dup

Reply:
There are no DROP TABLE for _dup tables?